### PR TITLE
Tpacket v3 3

### PR DESCRIPTION
**Pull request by regit**
Update of #1988. It fixes most comments done on previous PR. On the usability side, tpacket_v3 is now detected and build conditionally. It is activated by default on new configuration with a fallback on v2 if not available. Block size and timeout are now configurable. The default block size has been improved.

PR builds:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/155
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/151

### CODE REVIEW

File: suricata.yaml.in [hunk 1]
```diff
@@ -469,12 +469,22 @@ af-packet:
...
+    #tpacket-v3: yes
     # Ring size will be computed with respect to max_pending_packets and number
     # of threads. You can set manually the ring size in number of packets by setting
     # the following value. If you are using flow cluster-type and have really network
     # intensive single-flow you could want to set the ring-size independantly of the number
     # of threads:
     #ring-size: 2048
+    # Block size is used by tpacket_v3 only. It should set to a value high enough to contain
+    # a decent number of packets
+    #block-size: 32768
```
Comment (inliniac) [inline suricata.yaml.in]: size is in packets or bytes?

File: suricata.yaml.in [hunk 2]
```diff
@@ -469,12 +469,22 @@ af-packet:
...
+    # tpacket_v3 block timeout
+    #block-timeout: 10
```
Comment (inliniac) [inline suricata.yaml.in]: What does this mean?

File: suricata.yaml.in [hunk 3]
```diff
@@ -512,8 +522,9 @@ af-packet:
   # Put default values here
   - interface: default
     #threads: auto
-    #use-mmap: yes
+    use-mmap: yes
```
Comment (inliniac) [inline suricata.yaml.in]: could we make this default on the code level?

**Comment by inliniac** [comment]
Looks like we still default to v2 if the yaml doesn't specify v3 explicitly.

**Comment by inliniac** [comment]
Surprisingly, it apparently works fine to have an IPS with one iface in v2 and the other in v3 mode :)

Still something to block perhaps, sounds like a debugging nightmare to me.

File: suricata.yaml.in (hunk 1 above)
Comment (regit) [inline suricata.yaml.in]: bytes. Comment in yaml updated.

File: suricata.yaml.in (hunk 2 above)
Comment (regit) [inline suricata.yaml.in]: an opened block is passed to userspace if it is not filled in block-timeout milliseconds.

Comment added to yaml

**Pull request closed without merging**